### ISSUE TITLE
Issues reading WiFi SSID Characteristic

### ISSUE THREAD

**Issue by ryanteck** [NebraLtd/helium-miner-config#43]
A new issue has popped up where it is having difficulties reading the WiFi SSID characteristic.

Requires more investigation.

![image](https://user-images.githubusercontent.com/1104745/113442500-12151780-93e8-11eb-9c6d-4bb3b0996360.png)

**Comment by ryanteck** [comment]
I'm not quite sure why this is happening.

It seems to be intermittent, almost as if the Helium app times out much quicker than it used to.

**Comment by ryanteck** [comment]
It does seem to be that it's taking just that bit longer to respond to the stage where the Helium App isn't getting the response quick enough for it's liking.

**Comment by ryanteck** [comment]
```
root@60bf626:/opt/helium-miner-config# time nmcli device wifi
IN-USE  SSID                       MODE   CHAN  RATE        SIGNAL  BARS  SECURITY 
*       <redacted>               Infra  10    195 Mbit/s  67      ▂▄▆_  WPA2     
        <redacted>               Infra  1     54 Mbit/s   49      ▂▄__  --       
        <redacted>  Infra  10    130 Mbit/s  49      ▂▄__  WPA2     

real    0m5.432s
user    0m0.206s
sys     0m0.094s
```

**Comment by ryanteck** [comment]
Ok so the issue is:

In some cases it takes too long for NMCLI to respond back with the SSID list. Of which it seems the Helium app now does have a 5 Second timeout window.

This means it's intermittent as sometimes it will respond straight away, or take the few attempts to do so.

Sometimes it will succeed with responding the SSID, then lock up when responding all networks available.

Proposed solution:

Possibly a function or thread that calls the nmcli command to scan for networks or such instead.

If a funciton, the function could be called when advertising is first enabled ready for the first request.
When the device requests this data, the function can be called in the background. But if it takes too long for nmcli to respond it would respond with the already cached data.

Alternatvely could be a thread that starts alongside advertising being enabled, then when enabled checks every 5-10 seconds.

By moving it to either of the above, as a few commands share this data should be a smoother experience.

Will start integrating tomorrow.

**Comment by ryanteck** [comment]
Why hasn't it been picked up until now?

I noticed this only recently on my indoor hotspot when setting location at a later date, and @suprnrdy also noticed it.

I think what happened here is the resources consumed by the miner on initial sync meant that when in my case with the setting location later. Typically the setup is ran and these commands are issued without a network connection which means the miner isn't syncing. And Nmcli responds in ~ 3-4 Seconds

What happened is that when I then went to re-connect to set location, and the unit Parley had was initially connected to network enough for syncing to start is that Nmcli was just taking over 5 seconds to respond. Causing the issue.
